packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -1,4 +1,9 @@
-import { Directive, VNode, DirectiveBinding, warn } from '@vue/runtime-core'
+import {
+  DirectiveHooks,
```

Comment:
I don't think this file needs to be touched. The new `Directive` type works just fine.

Reply:
Reduced changes a bit, but I probably can't get rid of those at all.

Reply:
we could probably remove generics from these `DirectiveBinding`s and leave them as `any`.
![image](https://user-images.githubusercontent.com/19504461/66711632-2bf81000-ed98-11e9-968f-cf13b9992ef5.png)

Reply:
Ok in this case the previous change actually makes more sense. However I'm not super happy with the naming of the types. Let's adjust it to:

- `Directive` -> `ObjectDirective`
- `export type FunctionDirective<T = any> = DirectiveHook<T>`
- `export type Directive<T = any> = FunctionDirective<T> | ObjectDirective<T>`

And use `ObjectDirective` in this file.